### PR TITLE
Simple Config mode removed

### PR DESCRIPTION
**Pull request by libuweber**
Remove simple config mode

### CODE REVIEW

**Review by orltom — APPROVED** [review]
looks good to me

File: charts/microgateway/CHANGE-NOTES.md [hunk 1]
```diff
@@ -5,14 +5,13 @@
...
 
-- Advanced DSL Configuration is not supported anymore. Configurations using the Advanced DSL Mode will have to migrate to Expert DSL mode.
+- Advanced and Simple DSL Configuration are not supported anymore. Configurations using Advanced or Simple DSL Mode will have to migrate to the standard Microgateway DSL configuration. Please refer to the [Microgateway Documentation](https://docs.airlock.com/microgateway/2.0) for further information.
 - Helm Chart parameter 'config.generic.env' has been renamed to 'config.generic.configEnv'
+- The service name for the echo service has been changed from 'backend-service' to 'backend' to match the microgateway default value. The echo service name can be configured using 'echo-server.fullnameOverride'.
 
-#### Microgateway DSL
-
-The Expert DSL configuration mode makes use of the Micogateway DSL. Expert mode configurations therefore are affected by breaking changes in the microgateway DSL.
+#### Breaking Changes in the Microgateway DSL
```
Comment (dieti80) [inline charts/microgateway/CHANGE-NOTES.md:15]: Typo: DLS => DSL

File: charts/microgateway/CHANGE-NOTES.md [hunk 2]
```diff
@@ -5,14 +5,13 @@
 - Update to Microgateway 2.0.0.
 
 ### Breaking Changes
-###Ç Helm Chart Configuration
+#### Helm Chart Configuration
 
-- Advanced DSL Configuration is not supported anymore. Configurations using the Advanced DSL Mode will have to migrate to Expert DSL mode.
+- Advanced and Simple DSL Configuration are not supported anymore. Configurations using Advanced or Simple DSL Mode will have to migrate to the standard Microgateway DSL configuration. Please refer to the [Microgateway Documentation](https://docs.airlock.com/microgateway/2.0) for further information.
```
Comment (dieti80) [inline charts/microgateway/CHANGE-NOTES.md:10]: config.dsl.expert must be migrated to config.dsl

File: charts/microgateway/README.md.gotmpl [hunk 3]
```diff
@@ -402,14 +342,13 @@ extraVolumeMounts:
...
-          hostname: virtinc.com
-        mappings:
-          - name: virtinc
-            mapping_template_path: /config/template/mapping.xml
+  dsl:
+    apps:
+    - virtual_host:
+        hostname: virtinc.com
+      mappings:
+        - name: virtinc
```
Comment (dieti80) [inline charts/microgateway/README.md.gotmpl]: Has nothing to do with "removing simple mode" but we should remove "name: virtinc" here. If a mapping_template_path is used, the name comes from the xml.

File: charts/microgateway/README.md.gotmpl [hunk 4]
```diff
@@ -51,7 +52,6 @@ The current chart version is: {{ template "chart.version" . }}
     * [Certificates for Microgateway](#certificates-for-microgateway)
```
Comment (dieti80) [inline charts/microgateway/README.md.gotmpl:52]: I think config.generic.tlsSecretName is not required anymore. It has been deleted in configmap.yaml. Therefore the chapter "Certificates for Microgateway" should be deleted as well as config.generic.tlsSecretName.

With extraVolumes and extraVolumeMounts additional secrets can be mounted.

File: charts/microgateway/README.md.gotmpl [hunk 5]
```diff
@@ -402,14 +342,13 @@ extraVolumeMounts:
...
-          - name: virtinc
-            mapping_template_path: /config/template/mapping.xml
+  dsl:
+    apps:
+    - virtual_host:
+        hostname: virtinc.com
+      mappings:
+        - name: virtinc
+          mapping_template_path: /config/template/mapping.xml
 ```
```
Comment (dieti80) [inline charts/microgateway/README.md.gotmpl:351]: Typo: microgateway => Microgateway

https://github.com/ergon/airlock-helm-charts/blob/e987400a7bf194996bcd961d1a0114225fa3c881/charts/microgateway/README.md.gotmpl#L639

File: charts/microgateway/README.md.gotmpl (hunk 4 above)
Comment (dieti80) [reply charts/microgateway/README.md.gotmpl:52]: That would simplify the README, deployment.yaml and values.yaml :-)

**Review by dieti80 — APPROVED** [review]
Looks good to me. Just a few comments/improvements which could be done. Have a look and decide whether you think the suggested changes make sense.

File: charts/microgateway/README.md.gotmpl (hunk 4 above)
Comment (libuweber) [reply charts/microgateway/README.md.gotmpl:52]: I will do this in a separate pull request.

**Pull request merged**